src/components/Navbar/Navbar.test.tsx

```diff
@@ -0,0 +1,100 @@
+import { useState } from "react";
+import { IntlProvider } from "react-intl";
+import { BrowserRouter } from "react-router-dom";
+import { fireEvent, render, screen } from "@testing-library/react";
+import { messages } from "../../App/App.const";
+import Navbar from "./Navbar";
+import {
+  DarkModeContext,
+  DarkModeContextProvider,
+} from "../../contexts/DarkModeContext";
+
+describe("testing navbar component", () => {
+  function NavbarMock() {
+    const [search, setSearch] = useState("");
+    const [language, setLanguage] = useState("en");
+    return (
+      <IntlProvider
+        messages={messages[language as keyof typeof messages]}
+        locale={language}
+        defaultLocale="en"
+      >
+        <BrowserRouter>
+          <Navbar
+            setLanguage={setLanguage}
+            language={language}
+            setSearch={setSearch}
+            search={search}
+          />
+        </BrowserRouter>
+      </IntlProvider>
+    );
+  }
+
+  test("renders logo correctly", async () => {
+    render(<NavbarMock />);
```

Comment:
dobrze wykorzystujesz funkcję render, ale powinna ona renderować zaimportowany komponent 'Navbar' a nie taką konstrukcję, jaką wyżej stworzyłeś. Zdefiniowałeś jakiś komponent tutaj tak więc masz teraz w aplikacji tak naprawdę dwa takie same komponenty i jeśli zmieni się coś w komponencie 'tsx' to zmiany musiałbyś też wprowadzić tutaj. 
Generalnie działa to tak, że importujesz dany komponent i go renderujesz i na tym komponencie piszesz testy